src/Common/src/Internal/Runtime/EEType.cs

```diff
@@ -1220,22 +1221,35 @@ internal TypeManagerHandle TypeManager
         {
             get
             {
-                // This is always a pointer to a pointer to a type manager
-                return *(TypeManagerHandle*)_ppTypeManager;
+                IntPtr typeManagerIndirection;
+                if (IsDynamicType || !SupportsRelativePointers)
+                    typeManagerIndirection = GetField<Pointer>(EETypeField.ETF_TypeManagerIndirection).Value;
+                else
+                    typeManagerIndirection = GetField<RelativePointer>(EETypeField.ETF_TypeManagerIndirection).Value;
+
+                return *(TypeManagerHandle*)typeManagerIndirection;
             }
         }
 #if TYPE_LOADER_IMPLEMENTATION
         internal IntPtr PointerToTypeManager
         {
             get
             {
-                // This is always a pointer to a pointer to a type manager
-                return _ppTypeManager;
+                uint cbOffset = GetFieldOffset(EETypeField.ETF_TypeManagerIndirection);
+                fixed (EEType* pThis = &this)
```

Comment:
`fixed (EEType* pThis = &this)` is unnecessary. `Unsafe.AsPointer` should work just fine.

Reply:
Yep - this pattern is replicated in many places in this file. I'll fix those at some point too.